tests/test_affinity.py

```diff
@@ -1,5 +1,30 @@
 import receptor_affinity
+from receptor_affinity.exceptions import RouteUnavailableError, RouteMismatchError
+from receptor_affinity.mesh import Mesh, Node
 
 
 def test_version():
     assert hasattr(receptor_affinity, '__version__')
+
+
+def test_str_route_unavailable_error():
+    """Call ``str`` on a ``RouteUnavailableError``."""
+    node = Node('my node')
+    err = RouteUnavailableError(node)
+    str(err)
+
+
+def test_str_route_mismatch_error():
+    """Call ``str`` on a ``RouteMismatchError``."""
+    # Node.__init__ doesn't require all required attributes. It should be fixed, but in the
+    # meantime, using Node.create_from_config works around this issue, as it provides many default
+    # values.
+    #
+    # RouteMismatchError calls Node.get_routes, which requires a live receptor process with a
+    # working stats port. (Is that a good design decision?)
```

Comment:
I question if the Exception should go and do more work which could ultimately then fail. This could potentially lead us to strange tracebacks that are difficult to debug. Can we not imply the mismatches to report if we design the other exceptions to hold the right information?

Reply:
Yeah, this is a real design issue. I'm very much interested in better exception design.

Reply:
I dropped the calls to `get_routes()`, so the exceptions do less work, and failures are less likely. AFAICT, the `__str__` implementations now don't touch the receptor processes.